docs/source/plugins/index.rst

```diff
@@ -366,10 +366,66 @@ You should now have a running FiftyOne server and App, including your plugin.
     `yarn build`. You can setup a watcher to do this automatically. See
     `nodemon <https://www.npmjs.com/package/nodemon>`_.
 
-Publishing your plugin
-----------------------
+Plugin Files
+------------
 
-TBD
+The `fiftyone.yaml` file is used to define the plugin's metadata and operators.
+
+fiftyone.yaml (Plugin Metadata)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The plugin metadata is specified as a dictionary within the YAML file. The following fields are available for defining the plugin:
+
+- `name` (required): The name of the plugin. Usually in the form of `@org/plugin-name`.
+- `author`: The author or organization responsible for the plugin.
+- `version`: The version of the plugin. It should be a valid semantic version.
+- `license`: The license under which the plugin is distributed.
+- `description`: A brief description of the plugin.
+- `fiftyone`: A dictionary containing information about the compatibility of the plugin with FiftyOne.
+- `fiftyone.version`: A semver version range specifying the required FiftyOne version for the plugin to work properly.
+- `operators`: A list of operator names provided by the plugin.
+
+Optional Files
+~~~~~~~~~~~~~~
+
+The plugin directory may contain the following optional files:
+
+- `package.json`: A JSON file containing additional information about the plugin, including the JS bundle file path.
+- `__init__.py`: A Python entry file for the plugin.
+- `dist/index.umd.js`: A JS bundle file for the plugin.
+
+Publishing a FiftyOne Plugin
+----------------------------
+
+Publishing a plugin in FiftyOne allows you to share your custom functionality and operators with the community. This document outlines the steps required to publish a plugin by either committing the plugin files to a GitHub repository or uploading a zip file.
+
+Committing Plugin Files to GitHub
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To publish a plugin by committing the plugin files to a GitHub repository, follow these steps:
+
+1. Create a GitHub repository: Create a new repository on GitHub to host your plugin. Make sure to choose an appropriate name and provide a brief description of your plugin.
```

Comment:
is this necessary? I think the workflow would normally be:
1. create a new directory ideally named the same as the plugin's name
2. run `git init` inside 
3. push to github when ready, making sure plugin metadata file exists